File: web/lib/risk-registry.ts
// Mirrors x_elite/risk_registry.py exactly, so the CLI client and this web
// chat agree on which tools are SAFE vs which require a guardrail check.
// Deterministic lookup only - the model is never trusted to self-assess risk.

export type RiskTier = 'SAFE' | 'CONFIRM_REQUIRED';

export const TOOL_RISK: Record<string, RiskTier> = {
  get_board_status: 'SAFE',
  flash_heart: 'SAFE',
  trigger_alert: 'CONFIRM_REQUIRED',
};

// Unknown tool names default to the safest posture, not to SAFE.
export function riskTier(toolName: string): RiskTier {
  return TOOL_RISK[toolName] ?? 'CONFIRM_REQUIRED';
}

File: x_elite/risk_registry.py
# Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
# SPDX-License-Identifier: BSD-3-Clause
"""Static risk tiers for MCP tools.

Deterministic lookup only — the model is never trusted to self-assess the
risk of its own tool call. An unknown tool name defaults to the safest
posture (CONFIRM_REQUIRED) in x_elite/client.py, not to SAFE.
"""

TOOL_RISK = {
    "get_board_status": "SAFE",
    "flash_heart": "SAFE",
    "trigger_alert": "CONFIRM_REQUIRED",
    "list_branches": "SAFE",
    "delete_branch": "CONFIRM_REQUIRED",
    "wipe_repository": "CONFIRM_REQUIRED",
}
